### PR TITLE
fix: improve test speed and support win-arm64

### PR DESCRIPTION
**Pull request by baszalmstra**
### Description

I couldnt run some tests on my local arm64 windows machine because some packages are not available for arm64 (yet). 

### Solution

I added a platform to the toolcache to allow installing tools for other platforms than the current one. 

I also found that the toolcache was instantiated on each build invocation which effectively removes all caching. 

Finally I sped up the toolcache tests by using sharded repodata and changing the tool that was installed to something more lightweight.

@nichmor This touches some of your code, please review.

### CODE REVIEW

File: crates/pixi_config/src/lib.rs [hunk 1]
```diff
@@ -696,6 +696,23 @@ impl From<&Config> for rattler_repodata_gateway::ChannelConfig {
 }
 
 impl Config {
+    /// Constructs a new config that is optimized to be used in tests.
+    ///
+    /// This instance is optimized to provide the fastest experience for tests.
+    pub fn for_tests() -> Self {
```
Comment (nichmor) [inline crates/pixi_config/src/lib.rs:702]: could we use `#cfg[tests]` here?

**Review by nichmor — APPROVED** [review]

File: src/build/mod.rs [hunk 2]
```diff
@@ -255,16 +251,11 @@ impl BuildContext {
             }
         }
 
-        let tool_context = ToolContext::builder()
```
Comment (nichmor) [inline src/build/mod.rs:258]: 🤦

File: crates/pixi_config/src/lib.rs (hunk 1 above)
Comment (baszalmstra) [reply crates/pixi_config/src/lib.rs:701]: I want to expose this as a function from the crate. Also for other crates.

Comment (nichmor) [reply crates/pixi_config/src/lib.rs:701]: got it now

**Pull request merged**